### PR TITLE
Implement the logic for quorum check

### PR DESCRIPTION
**Pull request by vcastellm**
# Description

This PR goes together with https://github.com/0xPolygon/go-ibft/pull/39

Depending on the message type it will implement different kinds of quorum checks.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Local cluster can run fine.

### CODE REVIEW

**Comment by b00ris** [comment]
The solution looks good to me.

**Review by b00ris — APPROVED** [review]

**Review by b00ris — APPROVED** [review]

**Review by Kourin1996 — APPROVED** [review]

**Review by b00ris — APPROVED** [review]

**Review by Nemanja0x — APPROVED** [review]

File: consensus/ibft/consensus_backend.go [hunk 1]
```diff
@@ -103,11 +104,12 @@ func (i *backendIBFT) ID() []byte {
 	return i.currentSigner.Address().Bytes()
 }
 
-func (i *backendIBFT) MaximumFaultyNodes() uint64 {
+// Q: This is not being used in go-ibft
+func (i *backendIBFT) MaximumFaulty() uint64 {
 	return uint64(CalcMaxFaultyNodes(i.currentValidators))
 }
```
Comment (Stefan-Ethernal) [inline consensus/ibft/consensus_backend.go]: Remove it?
```suggestion
```

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM

File: consensus/ibft/consensus_backend.go (hunk 1 above)
Comment (vcastellm) [reply consensus/ibft/consensus_backend.go]: Done https://github.com/0xPolygon/polygon-edge/pull/837/commits/f7f7b532085845605c43b23f607ac4dde32275c8

**Pull request merged**